snuba/datasets/plans/single_storage.py

```diff
@@ -69,6 +70,16 @@ def process_and_run_query(
         return process_and_run_query(query, request_settings)
 
 
+def get_query_data_source(relational_source: RelationalSource) -> Table:
+    assert isinstance(relational_source, TableSource)
```

Comment:
Since RelationalSource is going away, could this just be

```suggestion
def get_query_data_source(relational_source: TableSource) -> Table:
```

Reply:
It cannot be done without removing RelationalSource because of nasty circular dependencies:
1) making this change means that all places where we call get_query_data_source need to have a Table
2) all those place take it from a storage attached to a schema `storage.get_schema().get_data_source()`
3) that is a RelationnalSource.
4) In order to change the return type of that method we need to make Schema depend on tables.py, which already depends on the schema.